rhombus-lib/rhombus/private/amalgam/port.rkt

```diff
@@ -79,6 +110,49 @@
     [() (open-output-string)]
     [(name) (open-output-string name)]))
 
+(define/method (Port.Input.peek_byte port [skip 0])
+  #:inline
+  #:primitive (peek-byte)
+  (peek-byte port skip))
+
+(define/method (Port.Input.peek_char port)
+  #:inline
+  #:primitive (peek-char)
+  (peek-char port))
+
+(define/method (Port.Input.read_byte port)
+  #:inline
+  #:primitive (read-byte)
+  (read-byte port))
+
+(define/method (Port.Input.read_bytes port amt)
+  #:inline
+  #:primitive (read-bytes)
+  (read-bytes amt port))
+
+(define/method (Port.Input.read_char port)
+  #:inline
+  #:primitive (read-char)
+  (read-char port))
+
+(define (coerce-read-result v)
+  (cond
+    [(string? v) (string->immutable-string v)]
+    [else v]))
+
+(define/method Port.Input.read_line
+  #:inline
+  #:primitive (read-line)
+  (case-lambda
+    [(port) (coerce-read-result (read-line port))]
+    [(port mode)
```

Comment:
An enum makes sense.  I'm not sure how to attach it to the Port namespace.  Or should this be in a different namespace?

Reply:
Oh, right — this is implemented in `.rkt`, so it can't actually use `enum`. I'd write a new annotation with `define-annotation-syntax`, add it into the `#:namespace-fields` part of `Input`, and then document it as written `enum`. (Not sure I got the details right there, but something like that.)

Reply:
Ok that makes sense.